### PR TITLE
sys-kernel/coreos-{kernel,firmware}: include microcode

### PR DESCRIPTION
**Pull request by ajeddeloh**
Add support for building microcode into the kernel

### CODE REVIEW

File: sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild [hunk 1]
```diff
@@ -15,8 +15,11 @@ if [[ ${PV} == 99999999* ]]; then
 	EGIT_REPO_URI="git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git"
 	KEYWORDS=""
 else
-	SRC_URI="mirror://gentoo/linux-firmware-${PV}.tar.gz"
-	KEYWORDS="amd64 arm64"
+	GIT_COMMIT="2eefafb2e9dcbafdf4b83d8c43fcd6b75fd4ac78"
+	SRC_URI="https://git.kernel.org/cgit/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-${GIT_COMMIT}.tar.gz -> ${P}.tar.gz
```
Comment (dm0-) [inline sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild]: Do we want to change the tar file's name to `coreos-firmware`?  Seems odd.

File: sys-kernel/coreos-kernel/coreos-kernel-4.14.13-r2.ebuild [hunk 2]
```diff
@@ -60,6 +61,11 @@ src_prepare() {
 	# Symlink to bootengine.cpio so we can stick with relative paths in .config
 	ln -sv "${ROOT}"/usr/share/bootengine/bootengine.cpio build/ || die
 	config_update 'CONFIG_INITRAMFS_SOURCE="bootengine.cpio"'
+
+	# include all intel and amd microcode files, avoiding the signatures
```
Comment (dm0-) [inline sys-kernel/coreos-kernel/coreos-kernel-4.14.13-r2.ebuild]: I don't think we want this happening on arm64.

File: sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild [hunk 3]
```diff
@@ -15,8 +15,11 @@ if [[ ${PV} == 99999999* ]]; then
...
+		mirror://gentoo/microcode_amd_fam17h.tar.gz
```
Comment (dm0-) [inline sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild]: The 17h microcode isn't being installed.

```
@@ -40,6 +40,10 @@
 ./lib/firmware/bnx2x/bnx2x-e1h-7.13.1.0.fw
 ./lib/firmware/ql2322_fw.bin
 ./lib/firmware/ql2100_fw.bin
+./lib/firmware/amd-ucode/
+./lib/firmware/amd-ucode/microcode_amd.bin
+./lib/firmware/amd-ucode/microcode_amd_fam16h.bin
+./lib/firmware/amd-ucode/microcode_amd_fam15h.bin
 ./lib/firmware/ql2500_fw.bin
 ./lib/firmware/acenic/
 ./lib/firmware/acenic/tg1.bin
@@ -67,9 +71,9 @@
 ./lib/firmware/tigon/tg3_tso5.bin
 ./lib/firmware/tigon/tg3.bin
 ./lib/firmware/cxgb4/
+./lib/firmware/cxgb4/t5fw-1.16.63.0.bin
 ./lib/firmware/cxgb4/t4fw.bin
-./lib/firmware/cxgb4/t6fw-1.16.45.0.bin
+./lib/firmware/cxgb4/t4fw-1.16.63.0.bin
 ./lib/firmware/cxgb4/t6fw.bin
 ./lib/firmware/cxgb4/t5fw.bin
-./lib/firmware/cxgb4/t5fw-1.16.45.0.bin
-./lib/firmware/cxgb4/t4fw-1.16.45.0.bin
+./lib/firmware/cxgb4/t6fw-1.16.63.0.bin
```

File: sys-kernel/coreos-kernel/coreos-kernel-4.14.13-r2.ebuild [hunk 4]
```diff
@@ -26,7 +26,8 @@ DEPEND="${RDEPEND}
 	sys-fs/e2fsprogs
 	sys-fs/mdadm
 	sys-fs/xfsprogs
-	>=sys-kernel/coreos-firmware-20160331-r1:=
+	sys-firmware/intel-microcode
```
Comment (dm0-) [inline sys-kernel/coreos-kernel/coreos-kernel-4.14.13-r2.ebuild]: This needs to be conditional on the architecture.

File: sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild [hunk 5]
```diff
@@ -15,8 +15,11 @@ if [[ ${PV} == 99999999* ]]; then
 	EGIT_REPO_URI="git://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git"
 	KEYWORDS=""
 else
-	SRC_URI="mirror://gentoo/linux-firmware-${PV}.tar.gz"
-	KEYWORDS="amd64 arm64"
+	GIT_COMMIT="2eefafb2e9dcbafdf4b83d8c43fcd6b75fd4ac78"
+	SRC_URI="https://git.kernel.org/cgit/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-${GIT_COMMIT}.tar.gz
```
Comment (dm0-) [inline sys-kernel/coreos-firmware/coreos-firmware-99999999.ebuild]: This still gives the source a different name than Gentoo uses.  It should be `linux-firmware-${PV}.tar.gz`.

**Comment by ajeddeloh** [comment]
updated to add iucode_tool in sdk-depends, rebased on top of 4.14.14 kernel

**Pull request merged**